app/models/manageiq/providers/nuage/network_manager/vsd_client.rb

```diff
@@ -0,0 +1,125 @@
+require 'rest-client'
+require 'rubygems'
+require 'json'
+module ManageIQ::Providers
+  class Nuage::NetworkManager::VsdClient
+    include Vmdb::Logging
+    def initialize(server, user, password)
+      @server = server
+      @user = user
+      @password = password
+      @rest_call = Rest.new(server, user, password)
+      connected, data = @rest_call.login
+      if connected
+        @enterprise_id = data
+        return
+      end
+      _log.error('VSD Authentication failed')
+    end
+
+    def get_enterprises
+      response = @rest_call.get(@server + '/enterprises')
+      if response.code == 200
+        if response.body == ''
+          _log.warn('No enterprises present')
+          return
+        end
+        return JSON.parse(response.body)
+      end
+      _log.error('Error in connection for server ' + @server.to_s + ' ' + response.code.to_s)
+    end
+
+    def get_domains
+      response = @rest_call.get(@server + '/domains')
+      if response.code == 200
+        if response.body == ''
+          _log.warn('No domains present')
+          return
+        end
+        return JSON.parse(response.body)
+      end
+      _log.error('Error in connection ' + response.code.to_s)
+    end
+
+    def get_zones
+      response = @rest_call.get(@server + '/zones')
+      if response.code == 200
+        if response.body == ''
+          _log.warn('No zones present')
+          return
+        end
+        return JSON.parse(response.body)
+      end
+      _log.error('Error in connection ' + response.code.to_s)
+    end
+
+    def get_subnets
+      response = @rest_call.get(@server + '/subnets')
+      if response.code == 200
+        if response.body == ''
+          _log.warn('No subnets present')
+          return
+        end
+        subnets = JSON.parse(response.body)
+        return subnets
+      end
+      _log.error('Error in connection ' + response.code.to_s)
+    end
+
+    def get_vports
+      response = @rest_call.get(@server + '/vports')
+      if response.code == 200
+        if response.body == ''
+          _log.warn('No vports present')
+          return
+        end
+        return JSON.parse(response.body)
+      end
+      _log.error('Error in connection ' + response.code.to_s)
+    end
+
+    def get_vms
+      response = @rest_call.get(@server + '/vms')
+      if response.code == 200
+        if response.body == ''
+          _log.warn('No VM present')
+          return
+        end
+        return JSON.parse(response.body)
+      end
+      _log.error('Error in connection ' + response.code.to_s)
+    end
+
+    def install_license(license_str)
+      license_dict = {}
+      license_dict['license'] = license_str
+      response = @rest_call.post(@server + '/licenses', JSON.dump(license_dict))
+      if response.code != 201
+        if response.body == 'The license already exists'
+          _log.error('license install failed')
+        end
+      end
+    end
+
+    def add_csproot_to_cms_group
```

Comment:
This doesn't seem to be called anywhere.

If it's not needed, take it out.  If you need it later, it be added later.  Better to not confuse this PR with things that don't belong here.